### DISCUSSION TITLE
Button OnClick or other events

### DISCUSSION THREAD

**Discussion by tiagozaccaro** [data-driven-forms/react-forms#1281] [Need help]
How would you implement a Button on Data Driven Forms?
and how would you implemente things like onClick or on Submit or other events on the schema?
How would you implement thinks like changing component states and be able to do thinks like show a modal on button click or call submit?

I need to make this to be able to create complex forms directly on schema, so clients can build there own forms.

thanks.

**Comment by rvsia** [comment]
@tiagozaccaro Hello,

you can simply add your custom components and define custom events there, that can be controlled via props in the schema.

[Read more about custom components](https://data-driven-forms.org/mappers/custom-mapper).
